File: lab_2/lab_5/src/kustova/natalia/Folder.java
package kustova.natalia;

import java.util.ArrayList;

public class Folder extends AbstractFileSystemNode {

    private ArrayList<FileSystemNode> arrayList;

    public Folder(String name, FileSystemNode ... fileSystemNodes) {
        setName(name);
        arrayList = new ArrayList<>(fileSystemNodes.length);
        for (FileSystemNode cureNode : fileSystemNodes) {
            ((AbstractFileSystemNode)cureNode).setParent(this);
            arrayList.add(cureNode);
        }
    }

    public String getChildNodes() {
        StringBuilder result = new StringBuilder();
        result.append("Child nodes: ");
        for (FileSystemNode curNode : arrayList) {
            result.append("\n" + curNode.getName());
        }
        return result.toString();
    }

    public FileSystemNode getChildNode(int indxNode) {
        return arrayList.get(indxNode);
    }
}

File: lab_3/lab_5/app/src/main/java/kustova/natalia/lab_5/Folder.kt
package kustova.natalia.lab_5

import java.util.*

class Folder(
    override var name: String?,
    override var parentElement: FileSystemNode? = null,
    override var path: String? = "",
    vararg fileSystemNodes: FileSystemNode?
) : FileSystemNode {

    private val arrayList: ArrayList<FileSystemNode>

    init {
        arrayList = ArrayList(fileSystemNodes.size)
        for (cureNode in fileSystemNodes) {
            (cureNode as FileSystemNode).parentElement = this
            arrayList.add(cureNode)
        }
    }

    fun getChildNodes() : String {
            val result = StringBuilder()
            result.append("Child nodes: ")
            for (curNode in arrayList) {
                result.append("""
                    ${curNode}
                """.trimIndent())
            }
            return result.toString()
        }

    fun getChildNode(indxNode: Int): FileSystemNode {
        return arrayList[indxNode]
    }
}